src/vec.rs

```diff
@@ -29,13 +34,7 @@ use core::{cmp::Ordering, fmt, hash, iter::FromIterator, mem::MaybeUninit, ops,
 /// assert_eq!(*vec, [7, 1, 2, 3]);
 /// ```
 pub struct Vec<T, const N: usize> {
-    // NOTE order is important for optimizations. the `len` first layout lets the compiler optimize
-    // `new` to: reserve stack space and zero the first word. With the fields in the reverse order
-    // the compiler optimizes `new` to `memclr`-ing the *entire* stack space, including the `buffer`
-    // field which should be left uninitialized. Optimizations were last checked with Rust 1.60
```

Comment:
I think this is not relevant anymore. The unsized type is always necessarily the last  so the optimization should take place.